### PR TITLE
Add presubmit jobs for kueue

### PR DESCRIPTION
**Pull request by ArangoGutierrez**
Add rudimentary jobs for CI testing.

Signed-off-by: Carlos Eduardo Arango Gutierrez <carangog@redhat.com>

### CODE REVIEW

File: config/jobs/kubernetes-sigs/kueue/kueue-presubmits.yaml [hunk 1]
```diff
@@ -0,0 +1,39 @@
+presubmits:
+  kubernetes-sigs/kueue:
+  - name: pull-kueue-build
+    always_run: true
+    skip_branches:
+    - ahg-g-patch-1
+    decorate: true
+    annotations:
+      testgrid-dashboards: sig-node-kueue
```
Comment (alculquicondor) [inline config/jobs/kubernetes-sigs/kueue/kueue-presubmits.yaml:9]: sig-scheduling

File: config/jobs/kubernetes-sigs/kueue/kueue-presubmits.yaml [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+      containers:
+      - image: golang:1.17
+        command:
+        - make
+        args:
+        - build
+  - name: pull-kueue-build-image
+    always_run: true
+    skip_branches:
+    - ahg-g-patch-1
```
Comment (alculquicondor) [inline config/jobs/kubernetes-sigs/kueue/kueue-presubmits.yaml:22]: we'll delete this branch instead

File: config/jobs/kubernetes-sigs/kueue/kueue-presubmits.yaml [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+      testgrid-dashboards: sig-node-kueue
+      testgrid-tab-name: build
+      description: "Build kueue binaries"
+    spec:
+      containers:
+      - image: golang:1.17
+        command:
+        - make
+        args:
+        - build
```
Comment (alculquicondor) [inline config/jobs/kubernetes-sigs/kueue/kueue-presubmits.yaml:18]: Can you also add `test`?

**Comment by alculquicondor** [comment]
You might also need to modify `config/tests/testgrids/config_test.go`

File: config/jobs/kubernetes-sigs/kueue/kueue-presubmits.yaml (hunk 1 above)
Comment (ArangoGutierrez) [reply config/jobs/kubernetes-sigs/kueue/kueue-presubmits.yaml:9]: mega face palm, sorry I took the template from the NFD project, fixing.....

File: config/jobs/kubernetes-sigs/kueue/OWNERS [hunk 4]
```diff
@@ -0,0 +1,8 @@
+# See the OWNERS docs at https://go.k8s.io/owners
+
+approvers:
+- ahg-g
+- alculquicondor
+reviewers:
+- ahg-g
+- alculquicondor
```
Comment (alculquicondor) [inline config/jobs/kubernetes-sigs/kueue/OWNERS:8]: add new line at the end

File: config/jobs/kubernetes-sigs/kueue/kueue-presubmits.yaml [hunk 5]
```diff
@@ -0,0 +1,49 @@
...
+      preset-dind-enabled: "true"
+    spec:
+      containers:
+      - image: gcr.io/k8s-staging-test-infra/kubekins-e2e:v20220216-aa6d36a90c-master
+        securityContext:
+          privileged: true
+        command:
+        - make
+        args:
+        - docker-build
```
Comment (alculquicondor) [inline config/jobs/kubernetes-sigs/kueue/kueue-presubmits.yaml:48]: If we already have this, maybe a presubmit for `make build` is redundant?

File: config/jobs/kubernetes-sigs/kueue/kueue-presubmits.yaml [hunk 6]
```diff
@@ -0,0 +1,49 @@
+presubmits:
+  kubernetes-sigs/kueue:
+  - name: pull-kueue-build
+    always_run: true
+    decorate: true
+    annotations:
+      testgrid-dashboards: sig-scheduling
```
Comment (alculquicondor) [inline config/jobs/kubernetes-sigs/kueue/kueue-presubmits.yaml:7]: maybe you have to use a different testgrid name?

File: config/jobs/kubernetes-sigs/kueue/OWNERS (hunk 4 above)
Comment (ArangoGutierrez) [reply config/jobs/kubernetes-sigs/kueue/OWNERS]: q: anyone else we could add?

File: config/jobs/kubernetes-sigs/kueue/kueue-presubmits.yaml [hunk 7]
```diff
@@ -0,0 +1,34 @@
...
+      - image: golang:1.17
+        command:
+        - make
+        args:
+        - test
+  - name: pull-kueue-build-image
+    always_run: true
+    decorate: true
+    annotations:
+      testgrid-dashboards: sig-scheduling-kueue-image
```
Comment (alculquicondor) [inline config/jobs/kubernetes-sigs/kueue/kueue-presubmits.yaml:21]: Can we do sig-scheduling-kueue for both?

Comment (ArangoGutierrez) [reply config/jobs/kubernetes-sigs/kueue/kueue-presubmits.yaml:21]: no, see last error
```bash
could not write config: 1 error occurred:
	* found duplicate name after normalizing: (DashboardTab) build
FAIL	k8s.io/test-infra/config/tests/testgrids	3.742s
```

Comment (alculquicondor) [reply config/jobs/kubernetes-sigs/kueue/kueue-presubmits.yaml:21]: but wasn't that because of testgrid-tab-name? (I didn't see the previous version).

Comment (ArangoGutierrez) [reply config/jobs/kubernetes-sigs/kueue/kueue-presubmits.yaml]: fixed! yay!

File: config/jobs/kubernetes-sigs/kueue/kueue-presubmits.yaml [hunk 8]
```diff
@@ -0,0 +1,36 @@
...
+      testgrid-tab-name: pull-kueue-test-master
+      description: "Run kueue tests"
+    spec:
+      containers:
+      - image: golang:1.17
+        command:
+        - make
+        args:
+        - test
+  - name: pull-kueue-build-image-master
```
Comment (alculquicondor) [inline config/jobs/kubernetes-sigs/kueue/kueue-presubmits.yaml:18]: pull-kueue-build-image-main

File: config/jobs/kubernetes-sigs/kueue/kueue-presubmits.yaml [hunk 9]
```diff
@@ -0,0 +1,36 @@
+presubmits:
+  kubernetes-sigs/kueue:
+  - name: pull-kueue-test-master
```
[the remark this answers is not in the document]
Comment (ArangoGutierrez) [reply config/jobs/kubernetes-sigs/kueue/kueue-presubmits.yaml:3]: sure

File: config/jobs/kubernetes-sigs/kueue/kueue-presubmits.yaml [hunk 10]
```diff
@@ -0,0 +1,64 @@
...
+      testgrid-tab-name: pull-kueue-test-integration-main
+      description: "Run kueue test-integration"
+    spec:
+      containers:
+      - image: golang:1.17
+        command:
+        - make
+        args:
+        - test-integration
+  - name: pull-kueue-verify-main
```
Comment (ArangoGutierrez) [inline config/jobs/kubernetes-sigs/kueue/kueue-presubmits.yaml:33]: @alculquicondor

Comment (ArangoGutierrez) [reply config/jobs/kubernetes-sigs/kueue/kueue-presubmits.yaml:33]: https://github.com/kubernetes-sigs/kueue/pull/21

**Review by alculquicondor — COMMENTED** [review]
/lgtm

**Comment by dims** [comment]
please assign to me when the CI jobs are green. thanks!

**Review by alculquicondor — COMMENTED** [review]
/lgtm

**Comment by ArangoGutierrez** [comment]
> please assign to me when the CI jobs are green. thanks!
test are green now, thanks
/assign dims

**Comment by ArangoGutierrez** [comment]
> please assign to me when the CI jobs are green. thanks!

test are green now, thanks
/assign dims

**Pull request merged**